openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMResultTest.java

```diff
@@ -0,0 +1,285 @@
+     *
+     * @throws URISyntaxException    In case of error retrieving the data resource path.
+     * @throws IOException           In case of error reading data.
+     * @throws ModelLoadingException In case of error training the model.
+     */
+    @Test
+    public void ensureLightgbmScoresAndContributions() throws URISyntaxException, IOException, ModelLoadingException {
+
+        final Dataset dataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath("treeshap_t/treeshap_train.csv"),
+                SCHEMA,
+                MAX_INSTANCES_TO_TRAIN
+        );
+        final LightGBMBinaryClassificationModel model = fit(
+                dataset,
+                MODEL_TRAINING_PARAMS,
+                CHUNK_SIZE_INSTANCES
+        );
+
+        final ArrayList<List<Double>> classScores = new ArrayList<>(2);
+        classScores.add(new LinkedList<>());
+        classScores.add(new LinkedList<>());
+        final ArrayList<List<Double>> classContributions = new ArrayList<>(2);
+        classContributions.add(new LinkedList<>());
+        classContributions.add(new LinkedList<>());
+
+        final Iterator<Instance> iterator = dataset.getInstances();
+
+        final Dataset resultDataset = CSVUtils.getDatasetWithSchema(
+                TestResources.getResourcePath("treeshap_t/treeshap_result.csv"),
+                RESULT_SCHEMA,
+                MAX_INSTANCES_TO_TRAIN
+        );
+
+        final Iterator<Instance> resultIterator = resultDataset.getInstances();
+
+        while (iterator.hasNext() && resultIterator.hasNext()) {
+            final Instance instance = iterator.next();
+            final Instance resultInstance = resultIterator.next();
+
+            final double[] scoreDistribution = model.getClassDistribution(instance);
+            final double[] featureContributions = model.getFeatureContributions(instance);
+
+            assertThat(scoreDistribution.length)
+                    .as("Class distribution should contain two values.")
+                    .isEqualTo(2);
+
+            final double prediction = resultInstance.getValue(9);
+            assertThat(isNearlyEquals(scoreDistribution[1], prediction))
+                    .as("The score should be expected.")
+                    .isTrue();
+
+            for (int i = 0; i < featureContributions.length; i++) {
+                assertThat(isNearlyEquals(featureContributions[i], resultInstance.getValue(i)))
+                        .as("The contribution should be expected.")
+                        .isTrue();
+            }
+        }
+    }
+
+    /**
+     * Compares two double values to a precision of 16 decimal places.
+     *
+     * @param d1 The first double value.
+     * @param d2 The second double value.
+     * @return {@code true} if values are nearly equal, otherwise {@code false}.
+     */
+    private boolean isNearlyEquals(final double d1, final double d2) {
```

Comment:
why do we need this?

can we replace this with something like this?
```
        assertThat(1d)
                .isCloseTo(1, Offset.offset(1.0e-6));
```

Reply:
it didn't show up at the time, aka I forgot its existence :smile: